mlflow/models/utils.py

```diff
@@ -950,6 +979,12 @@ def _is_scalar(x):
             )
     if input_schema.is_tensor_spec():
         return _enforce_tensor_schema(pf_input, input_schema)
+    elif isinstance(original_pf_input, SparkDataFrame):
+        if input_schema.has_input_names():
+            _enforce_named_col_schema(pf_input, input_schema)
+        else:
+            _enforce_unnamed_col_schema(pf_input, input_schema)
+        return original_pf_input
```

Comment:
Instead if adding the special case for Spark DF here, can we update `_enforce_named_col_schema` and `_enforce_unnamed_col_schema` to return directly if passed a Spark DataFrame? Can we also add a note explaining why?

Reply:
This will move the logic of converting from a spark df to a pandas df into the `_enforce_named_col_schema` and `_enforce_unnamed_col_schema` helper functions. Not sure if this is ideal since the it made add bulk to those helper functions.

Reply:
It may be problematic to drop columns from the dataframe that are not included in the signature because currently feature store signatures do not put arrays and maptypes into the signature. So, adding this feature will cause incompatibility with previously logged feature store models.

Reply:
Will change to not enforce schema for map types and array types in feature store models to allow for compatibility with perviously logged feature store models. For any other pyspark models will support these types.

Reply:
Added this feature